exporter/kafkaexporter/go.mod

```diff
@@ -1,6 +1,6 @@
 module github.com/open-telemetry/opentelemetry-collector-contrib/exporter/kafkaexporter
 
-go 1.21.0
+go 1.21.9
```

Comment:
```suggestion
go 1.21.0
```
I believe we want to keep this at its original value. [Related issue and references](https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/32356).

Reply:
From my understanding, the `go 1.21.X` specifies the _minimum_ `go` version that can be used to build and run this component. However, when building and releasing the collector, we're actually using [`go 1.22` from what I can tell](https://github.com/open-telemetry/opentelemetry-collector-releases/blob/d2c96926b8b02e370d420df625dbc47cfbafa1cc/.github/workflows/base-ci-goreleaser.yaml#L54), so this vulnerability wouldn't be relevant in the release.

Also, specifying a higher minimum version would require all packages that depend on this component to update their own go versions as well, which would create a lot of extra changes required.

I think we should revert.

Reply:
Done